pkg/archive/archive.go

```diff
@@ -382,6 +382,56 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 	return nil
 }
 
+// Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
```

Comment:
I'd rather do that once we are happy and have the stable code in place, and all tests passing, with no impact on performance, maybe as a follow up PR?